backend/btrixcloud/operator.py

```diff
@@ -1137,6 +1200,71 @@ async def do_crawl_finished_tasks(
         # finally, delete job
         await self.delete_crawl_job(crawl_id)
 
+    async def compute_execution_time(self, crawl_id, oid, crawl=None):
+        """Compute execution time for crawl from start and end times in redis"""
+        redis = None
+
+        print("Computing execution time", flush=True)
+
+        try:
+            redis_url = self.get_redis_url(crawl_id)
+            redis = await self._get_redis(redis_url)
+            if not redis:
+                return
+
+            scale = 2
+            if crawl:
+                scale = crawl.scale
+
+            execution_time = 0.0
+
+            for i in range(scale):
+                name = f"crawl-{crawl_id}-{i}"
+
+                start_times = await redis.lrange(f"{crawl_id}:start:{name}", 0, -1)
+                end_times = await redis.lrange(f"{crawl_id}:end:{name}", 0, -1)
+
+                print(f"Crawler {name} start times:", flush=True)
+                print(start_times, flush=True)
+
+                print(f"Crawler {name} end times:", flush=True)
+                print(end_times, flush=True)
+
+                for i, start_time_str in enumerate(start_times):
+                    start_time = from_timestamp_str(start_time_str)
+                    try:
+                        end_time = from_timestamp_str(end_times[i])
+                    except IndexError:
+                        # pylint: disable=line-too-long
+                        print(
+                            f"Start time {start_time_str} has no corresponding end time, using current time",
+                            flush=True,
+                        )
+                        end_time = datetime.now()
+
+                    duration = end_time - start_time
+                    seconds = duration.total_seconds()
```

Comment:
Perhaps we just store `Date.now()` in the crawler and then compare it here to avoid the timestamp conversion, and use `int(time.time() * 1000)` as the python equivalent?